### DISCUSSION TITLE
Easy tutorial to compile AVX build without AES-NI?

### DISCUSSION THREAD

**Discussion by Unknown78** [Alex313031/Thorium-Win#205] [Q&A]
As per issue on #180 
> I've looked up the keyword aes in the repo, and it spread across [many files](https://github.com/search?q=repo%3AAlex313031%2Fthorium+aes&type=code), I don't know which one to modify.
There's no tutorial of it on [BUILDING_WIN.md](https://github.com/Alex313031/thorium/blob/main/docs/BUILDING_WIN.md) either...

**Comment by Unknown78** [comment]
Hi @gz83, you're known as a good builder. Might give me some insight?

**Reply by gz83** [reply]
I think you can first try to find out whether there are parameters such as aes in the two files //build/config/compiler/BUILD.gn and //build/config/win/BUILD.gn. If so, please try to comment or remove them, and then start compiling.

If you have any other questions, you can send them here and maybe I can help you.

**Reply by Unknown78** [reply]
@gz83 I searched up with the keyword of "+aes" and could only found one thing that might be relevant for windows build.
https://github.com/Alex313031/thorium/blob/main/src/build/config/compiler/BUILD.gn#L1098-L1100

```gn
  if (use_avx) {
    rustflags += [ "-Ctarget-feature=+sse3,+ssse3,+sse4.1,+sse4.2,+pclmulqdq,+aes,+avx", ]
  }
```

Is this correct, that it is the only one?

**Reply by gz83** [reply]
Looks basically correct, but I'd also like to confirm your processor model (the one without AES support)

**Reply by Unknown78** [reply]
@gz83  as per  https://github.com/Alex313031/Thorium-Win/issues/180
It is Sandy Bridge i3-2310M
![image](https://github.com/Alex313031/Thorium-Win/assets/25132599/21985381-892f-4135-9ebd-bfd1ba3e7d60)

---
It seems like I don't have enough disk space needed to build Thorium.

After looking around in the internet, I found this code to run with online rust compiler.

```rust
use std::process::{Command};

fn main() {
    let output = Command::new("rustc")
        .arg("--print=cfg")
        .arg("-C")
        .arg("target-cpu=sandybridge")
        .arg("--target=x86_64-pc-windows-msvc")
        .output()
        .expect("Failed to execute rustc command");

    println!("{}", String::from_utf8_lossy(&output.stdout));
}
```

And obtain this list
```
debug_assertions
panic="unwind"
target_abi=""
target_arch="x86_64"
target_endian="little"
target_env="msvc"
target_family="windows"
target_feature="avx"
target_feature="cmpxchg16b"
target_feature="fxsr"
target_feature="pclmulqdq"
target_feature="popcnt"
target_feature="sse"
target_feature="sse2"
target_feature="sse3"
target_feature="sse4.1"
target_feature="sse4.2"
target_feature="ssse3"
target_feature="xsave"
target_feature="xsaveopt"
target_has_atomic="128"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_os="windows"
target_pointer_width="64"
target_vendor="pc"
windows
```

---

P.S. A switch to also bypass webgl2 blocklist is also nice...

Is there anything missing?

**Reply by gz83** [reply]
Thank you for your information. Your device is not powerful enough to compile Chromium. If you really need to try to compile, please consider changing to a more powerful device or try to rent a bare metal server (on the cloud)

Also sorry, Alex probably won't make a version without aes.

**Reply by SamratBarai** [reply]
Can I build it using the free tier in Lightning AI, since they provide you with 32 core cpus and 64 gigs ram etc
